src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SupplierAssessmentService.kt

```diff
@@ -63,6 +65,80 @@ class SupplierAssessmentService(
     return appointment
   }
 
+  fun scheduleNewSupplierAssessmentAppointment(
+    referralId: UUID,
+    durationInMinutes: Int,
+    appointmentTime: OffsetDateTime,
+    createdByUser: AuthUser,
+    appointmentDeliveryType: AppointmentDeliveryType,
+    appointmentDeliveryAddress: AddressDTO? = null,
+    npsOfficeCode: String? = null,
+  ): Appointment {
+
+    val sentReferral = referralRepository.findByIdAndSentAtIsNotNull(referralId) ?: throw EntityNotFoundException("Sent Referral not found [referralId=$referralId]")
+    val supplierAssessment = sentReferral.supplierAssessment ?: run {
+      SupplierAssessment(
+        id = UUID.randomUUID(),
+        referral = sentReferral,
+      )
+    }
+    supplierAssessment.currentAppointment?.let {
+      latestAppointment ->
+      if (latestAppointment.appointmentTime.isAfter(appointmentTime)) {
+        throw ResponseStatusException(HttpStatus.BAD_REQUEST, "can't schedule new supplier assessment appointment; new appointment occurs before previously scheduled appointment [referralId=$referralId]")
+      }
+      latestAppointment.appointmentFeedbackSubmittedAt ?: throw ResponseStatusException(HttpStatus.BAD_REQUEST, "can't schedule new supplier assessment appointment; existing appointment has no feedback delivered [referralId=$referralId]")
+    }
+    val appointment = appointmentService.scheduleNewAppointment(
+      sentReferral.id,
+      SUPPLIER_ASSESSMENT,
+      durationInMinutes,
+      appointmentTime,
+      createdByUser,
+      appointmentDeliveryType,
+      // Supplier assessment is always ONE_TO_ONE
+      AppointmentSessionType.ONE_TO_ONE,
```

Comment:
(Thinking about how to make the comment enforceable in code/regressions)

Feels like this is a good thing to encode into the preconditions of `scheduleNewAppointment` with `if appointmentType == SUPPLIER_ASSESSMENT and appointmentDeliveryType != ONE_TO_ONE then Error` and create a test for it

Reply:
Agreed. I've added this validation into SupplierAsessmentService